### PR TITLE
pulsar integration

### PR DESCRIPTION
**Pull request by daveads**
Pulsar integration Feature #583 
 
 https://github.com/permitio/opal/issues/583

### CODE REVIEW

**Comment by daveads** [comment]
https://github.com/user-attachments/assets/16ffac8b-091b-4947-ad76-5a36b26b4e1d

File: broadcaster/_backends/pulsar.py
Comment (danyi1212) [inline broadcaster/_backends/pulsar.py:57]: Same and clearer as `asyncio.to_thread()` 
https://docs.python.org/3/library/asyncio-task.html#asyncio.to_thread

Then also this wrapping method becomes obsolete, so better use `asyncio.to_thread()` on each place (better typing and clearity)

File: broadcaster/_backends/pulsar.py
Comment (danyi1212) [inline broadcaster/_backends/pulsar.py:51]: Better wrap everything inside this while in a try except all, so that an exception will not crash the consumer.

File: broadcaster/_backends/pulsar.py
Comment (danyi1212) [inline broadcaster/_backends/pulsar.py:50]: How does the shutdown procedure work? Based on a `CancelledError` on the consumer task?

File: broadcaster/_backends/pulsar.py
Comment (danyi1212) [inline broadcaster/_backends/pulsar.py:5]: Relative imports are dangerous, use absolute imports

**Review by danyi1212 — CHANGES_REQUESTED** [review]

File: broadcaster/_backends/pulsar.py
Comment (daveads) [reply broadcaster/_backends/pulsar.py:57]: okay

File: broadcaster/_backends/pulsar.py
Comment (daveads) [reply broadcaster/_backends/pulsar.py:5]: okayy

**Comment by daveads** [comment]
@danyi1212 ...

**Review by danyi1212 — APPROVED** [review]

**Comment by daveads** [comment]
@danyi1212   waiting for workflow approval

**Comment by daveads** [comment]
@danyi1212  
* Used `anyio.to_thread.run_sync()` which works across various Python versions, including 3.8
* Setup workflow for pulsar test on github action... 

Everything works fine currently on a test branch...

**Pull request merged**
